docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+}
+```
+
+To implement these methods, you can use the [testcafe-browser-tools](https://github.com/DevExpress/testcafe-browser-tools) library.
+
+For details, see [Browser Provider Methods](browser-provider-methods.md).
+
+### Example
+
+The following example code demonstrates how you can implement the simple browser provider for portable browsers.
+
+```js
+import path from 'path';
+import browserTools from 'testcafe-browser-tools';
+
+export default {
+    isMultiBrowser: true,
+
+    async openBrowser (id, pageUrl, browserName) {
+        var browserInfo = {};
+
+        switch (browserName) {
+            case 'firefox':
+                browserInfo.path = path.join(process.env.PORTABLE_BROWSERS_PATH, 'FirefoxPortable/FirefoxPortable.exe');
+                browserInfo.cmd  = '-new-window';
+                break;
+            case 'chrome':
+                browserInfo.path = path.join(process.env.PORTABLE_BROWSERS_PATH, 'GoogleChromePortable/GoogleChromePortable.exe');
+                browserInfo.cmd  = '--new-window';
+                break;
+            default:
+                throw new Error('Unsupported browser!');
+        }
+
+        await browserTools.open(browserInfo, pageUrl);
+    },
+
+    async closeBrowser (id) {
+        await browserTools.close(id);
+    },
+
+    async getBrowserList () {
+        return ['firefox', 'chrome'];
+    },
+
+    async isValidBrowserName (browserName) {
+        var browserList = await this.getBrowserList();
+
+        return browserList.indexOf(browserName) > -1;
+    },
+
+    async resizeWindow (id, width, height, currentWidth, currentHeight) {
+        await browserTools.resize(id, currentWidth, currentHeight, width, height);
+    },
+
+    async takeScreenshot (id, screenshotPath) {
+        await browserTools.screenshot(id, screenshotPath);
+    },
+};
+```
+
+> To learn how to implement provider methods, you can also explore the existing browser provider plugins, for example, [testcafe-browser-provider-saucelabs](https://github.com/DevExpress/testcafe-browser-provider-saucelabs/) or [testcafe-browser-provider-phantomjs](https://github.com/DevExpress/testcafe-browser-provider-phantomjs).
+
+## Building the Provider
+
+You can build a browser provider project by using the `build` Gulp task.
+
+```bash
+gulp build
+```
+
+## Using the Provider Development Version
+
+If you are still developing the browser provider, but need to test it within TestCafe, there is no need to publish the browser provider package to npm.
+You can link the browser provider to TestCafe by using the [npm link](https://docs.npmjs.com/cli/link) command.
+This allows you to work on the browser provider project and test it iteratively without having to re-publish everytime you make a change.
+
+To link the browser provider package, navigate to the provider directory and run `npm link`:
+
+```bash
+cd my-provider
+npm link
+```
+
+After that TestCafe will use the provider version you are currently developing.
```

Comment:
It's worth to mention that it will work only if `testcafe` was installed globally.